src/idl_gen_cpp.cpp

```diff
@@ -318,6 +312,22 @@ class CppGenerator : public BaseGenerator {
     }
   }
 
+  // generates C++0x or C++11 constructors
+  std::string GenConstructor(const std::string &type_name,
+                             const std::string &params) {
+      std::string o = parser_.opts.cpp_variant == IDLOptions::Cpp0x ? "(" : "{";
```

Comment:
maybe we can have a struct that contains all the different tokens between the C++ variants, this will cut down on the number of conditionals we'll need now and in the future. See for example `struct LanguageParameters` in idl_gen_general.cpp as an example.